drivers/counter/Kconfig.gd32

```diff
@@ -6,7 +6,7 @@
 config COUNTER_TIMER_GD32
 	bool "GD32 timer counter driver"
 	default y
-	depends on DT_HAS_GD_GD32_TIMER_ENABLED && SOC_FAMILY_GD32_ARM
```

Comment:
The timer has not yet worked with GD32V.
Please keep the restriction for ARM only

Reply:
Does that mean GD32V has a different IP? If so, GD32V dts files should have another compatible, ie, not `gd,gd32-timer`. For now I'll change to `!SOC_SERIES_GD32VF103`, because `SOC_FAMILY_GD32_ARM` is gonna disappear with hwmv2.

Reply:
The timer driver depends on interrupt controller behavior. It works correctly on ARM, but it needs improvement for RISC-V.
It's still not going well.

Reply:
Hi @soburi ,

Would you mind to open an issue and describe exactly what is the problem and the possible root of cause?

Reply:
I've got it. please wait a moment.